docs/clustering.rst

```diff
@@ -0,0 +1,224 @@
+default TCP port). Ensure that you had the root password for all machines at
+hand.
+
+Adding a peer
+^^^^^^^^^^^^^^
+
+We'll start by peering master01 and master02. To begin we need to ensure that
+passwordless SSH connection is possible between the nodes. Merlin includes
+a convince script to setup and install SSH keys across the nodes, that we'll
+use below.
+
+.. code-block:: none
+
+   [root@master01 ~]# mon sshkey push IP-OF-MASTER01
+   [root@master02 ~]# mon sshkey push IP-OF-MASTER02
+
+With the above, we should be able to SSH between both nodes, both as the
+``root`` user and the ``naemon`` user.
+
+Now we setup the peers in each nodes Merlin configuration. We'll use another
+``mon`` tool adjust Merlins configuration file.
+
+We'll start by adding master01 to master02's configuration.
+
+Start by adding add master01 to the Merlin config on master02.
+
+.. code-block:: none
+
+   [root@master02 ~]# non node add master01 type=peer address=IP-OF-MASTER02
+
+On master01 then, add master02 to the Merlin config, and push the Naemon
+configuration to master02. We always push configuration as the ``naemon`` user.
+Finally, restart Naemon & Merlin.
+
+.. code-block:: none
+
+   [root@master01 ~]# mon node add master02 type=peer address=IP-OF-MASTER02
+   [root@master01 ~]# su naemon -c "mon oconf push master02"
+   [root@master01 ~]# mon restart
+
+
+At this point we should have a healthy cluster, and we can use a few more
+``mon`` tools to look at our cluster state. These two nodes are now
+loadbalanced sharing the check executions equally.
+
+.. code-block:: none
+
+   [root@master01 ~]# mon node status
+   Total checks (host / service): 4 / 21
+
+   #00 0/1:1 local ipc: ACTIVE - 0.000s latency
+   Uptime: 15m 47s. Connected: 15m 48s. Last alive: 8s ago
+   Host checks (handled, expired, total)   : 2, 0, 4 (50.00% : 50.00%)
+   Service checks (handled, expired, total): 11, 0, 21 (52.38% : 52.38%)
+
+   #01 1/1:1 peer master02: ACTIVE - 0.000s latency - (UNENCRYPTED)
+   Uptime: 15m 48s. Connected: 15m 48s. Last alive: 8s ago
+   Host checks (handled, expired, total)   : 2, 0, 4 (50.00% : 50.00%)
+   Service checks (handled, expired, total): 10, 0, 21 (47.62% : 47.62%)
+
+   [root@master01 ~]# mon node tree
+    +-----+    +----------+
+    | ipc |----| master02 |
+    +-----+    +----------+
+
+Adding a poller
+^^^^^^^^^^^^^^^^
+
+Now that we have two peers in the master peergroup, we can add a poller. We must
+first decide which hostgroup(s) the poller should be responsible for. Before
+getting started with the setup, ensure that the hostgroup has already been added
+to the masters Naemon configuration. In our example we'll use a hostgroup called
+``pollergroup``.
+
+SSH connection must be established to both masters, so we start by adding SSH
+keys.
+
+.. code-block:: none
+
+   [root@poller01 ~]# mon sshkey push IP-OF-MASTER01
+   [root@poller01 ~]# mon sshkey push IP-OF-MASTER02
+   [root@master01 ~]# mon sshkey push IP-OF-POLLER01
```

Comment:
These lines look so similar (I had a hard time distinguishing the hostnames in the prompt even), so it's hard to read them and see the differences. Perhaps they could be grouped into per-node commands or something?

The same applies to other similar blocks with commands.

Reply:
I added a blank line so that commands from different systems are grouped together. Makes it a bit better at least.